### PR TITLE
Belindas-closet-nextjs-5-60-finish-web-user-auth

### PR DESCRIPTION
**Pull request by taylorpapke**
resolves: https://github.com/SeattleColleges/belindas-closet-nextjs/issues/60

to test:

- create a Mongo DB database and create a database and collection
- create a .env file under project folder (same level as .env.sample) and add the following:
MONGODB_URI=<your Mongo DB connection string w/ database and collection appended to the end>
NEXTAUTH_URL=http://localhost:3000/
NEXTAUTH_SECRET=secret
259872560-2fa9877b-e891-46fd-9cc9-8cef37ac01a7

- remember to replace username and password with your actual username and password from mongodb when you created your cluster.

sign up:
Screenshot 2023-09-01 at 3 46 43 PM
Screenshot 2023-09-01 at 3 47 15 PM

sign in:
Screenshot 2023-09-01 at 3 47 48 PM
Screenshot 2023-09-01 at 3 48 19 PM

### CODE REVIEW

File: app/api/auth/[...nextauth]/route.ts [hunk 1]
```diff
@@ -27,26 +25,37 @@ const authOptions: NextAuthOptions = {
...
                   password: string;
               };
-              
-              // perform login logic to find user from db
-              // upon capturing email and password, start and connect db to check for matching user email
+
               await startDb();
               const user = await UserModel.findOne({ email });
+              if (!user) {
+                  throw Error("email/password mismatch");
```
Comment (bcko) [inline app/api/auth/[...nextauth]/route.ts:32]: What happens when this error gets thrown? Do we have a place to catch & handle this error?

**Review by bcko — APPROVED** [review]

File: app/api/auth/[...nextauth]/route.ts (hunk 1 above)
Comment (bcko) [reply app/api/auth/[...nextauth]/route.ts:32]: You don't have to make changes in this pull request but let's create a ticket so we can talk about error handling practices

Comment (theGaryLarson) [reply app/api/auth/[...nextauth]/route.ts:32]: What I've seen done in other implementations is to implement a state isError, for example. Then above the throw method set isError to true. Then you will be able to use that state to render an error component in the return

**Review by theGaryLarson — APPROVED** [review]
Looks good. We will discuss error handling when we pick this back up during the school year.

**Pull request merged**